### PR TITLE
Merge entity guess function from PropertyRegsiter with EntityIdValueImpl

### LINKED ISSUE #424: Code duplication: determine an entity type by looking at the format of its id
There are at least two places in our code where we try to guess the type of an entity based on the format of its id:

https://github.com/Wikidata/Wikidata-Toolkit/blob/ac2692ac35e3ae7fc8148f416322e1e6f07beca0/wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java#L134-L162

and

https://github.com/Wikidata/Wikidata-Toolkit/blob/d76e8ee421856f7aba6bff6cf1f8ab13e03f451d/wdtk-rdf/src/main/java/org/wikidata/wdtk/rdf/PropertyRegister.java#L199-L229

That is a problem: it is easy to forget to update one of them (it caused #407).
We should expose this functionality publicly in the `wdtk-datamodel` module. This raises questions:
* where should this static method live?
* what representation of the entity types should it return? Currently one uses the JSON strings, the other uses IRIs for datatypes. I think it should be IRIs for entity types since that is what the `EntityIdValue` interface offers… But then we probably need other static methods to help convert between all these formats? It's a bit of a mess :-/

### PR DESCRIPTION
**Pull request by snehasi**
Resolves #424

### CODE REVIEW

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java [hunk 1]
```diff
@@ -143,33 +144,47 @@ public static EntityIdValue fromId(String id, String siteIri) {
...
+					return returnJsonEntity ? JSON_ENTITY_TYPE_FORM : DatatypeIdValue.DT_FORM;
 				} else if(id.contains("-S")) {
-					return JSON_ENTITY_TYPE_SENSE;
+					return returnJsonEntity ? JSON_ENTITY_TYPE_SENSE : DatatypeIdValue.DT_SENSE;
 				} else {
-					return JSON_ENTITY_TYPE_LEXEME;
+					return returnJsonEntity ? JSON_ENTITY_TYPE_LEXEME : DatatypeIdValue.DT_LEXEME;
 				}
 			case 'M':
 				return JSON_ENTITY_TYPE_MEDIA_INFO;
```
Comment (Tpt) [inline wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java:166]: You probably want to return here `DatatypeIdValue.DT_MEDIA_INFO`, no?

File: wdtk-rdf/src/main/java/org/wikidata/wdtk/rdf/PropertyRegister.java [hunk 2]
```diff
@@ -220,21 +221,9 @@ public String getPropertyUriPattern(PropertyIdValue propertyIdValue) {
...
-			if (value.getId().contains("F")) {
-				return DatatypeIdValue.DT_FORM;
-			} else if(value.getId().contains("S")) {
-				return DatatypeIdValue.DT_SENSE;
-			}
-			return DatatypeIdValue.DT_LEXEME;
-		case 'M':
-				return DatatypeIdValue.DT_MEDIA_INFO;
-		default:
+		try{
```
Comment (Tpt) [inline wdtk-rdf/src/main/java/org/wikidata/wdtk/rdf/PropertyRegister.java:224]: Usually we leave a space before curly braces (e.g; `try {`)

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java [hunk 3]
```diff
@@ -143,33 +144,47 @@ public static EntityIdValue fromId(String id, String siteIri) {
...
 			case 'Q':
-				return JSON_ENTITY_TYPE_ITEM;
+				return returnJsonEntity ? JSON_ENTITY_TYPE_ITEM : DatatypeIdValue.DT_ITEM;
 			default:
 				throw new IllegalArgumentException("Entity id \"" + id + "\" is not supported.");
 		}
 	}
 
+	/**
+	 * RReturns the entity type of the id like "item" or "property"
```
Comment (Tpt) [inline wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java:177]: Two `R`

File: wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java (hunk 1 above)
Comment (snehasi) [reply wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/EntityIdValueImpl.java]: Yes, thank you! I've fixed it and pushed my changes.

**Pull request merged**